### PR TITLE
Add new-homepage E2E-spec

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Changes: Added `new-homepage.cypress.spec.js`
- Not a bug-fix
- Team is VA Applications Shared Support
- No feature-flag used

## Related issue(s)
[n/a; QA-internal initiative]

## Testing done

- Ran spec locally.  Test results [successfully pushed to TestRail](https://dsvavsp.testrail.io/index.php?/runs/overview/56) [see Completed > Friday, December 16, 2022 section].  [See also screenshot below.]
image

NOTE: First test AXE-checks the entire homepage.  Subsequent tests do not need repeat AXE-checks.

## Screenshots
[n/a; not component work]

## What areas of the site does it impact?
[none; just adding an automated test here]

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

None; just your regular code review(s).

### CODE REVIEW

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+    cy.get('.homepage-common-tasks').should('exist');
+    cy.get('.homepage-blog').should('exist');
+    cy.get('.homepage-benefits-row').should('exist');
+    cy.get('.homepage-email-updates-signup').should('exist');
+
+    cy.injectAxeThenAxeCheck();
+  });
+
+  describe('Hero section', () => {
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:63]: AXE-check already done in first test [line 59] for entire page.  No need to repeat.

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+          cy.location('pathname').should(
+            'eq',
+            '/resources/the-pact-act-and-your-va-benefits/',
+          );
+        });
+
+      // skipping AXE-check -- already done in previous test
+    });
+
+    // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:96]: AXE-check already done in first test [line 59] for entire page.  No need to repeat.

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+
+    describe('Search subsection', () => {
+      describe('Search box', () => {
+        const keyword = 'health';
+        const suggestedKeyword = 'health benefits';
+        const checkResultsPage = searchTerm => {
+          cy.location('pathname').should('eq', '/search/');
+          cy.location('search').should(
+            'eq',
+            `?query=${searchTerm.replace(' ', '%20')}&t=false`,
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js]: Just pushed a fix.

[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js]: Now using replaceAll() instead of replace().

**Comment by jilladams** [comment]
@tlei123 we all crossed in timespace - Public Websites did a SPIKE (https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11704) and cut a ticket for implementing E2E tests, https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11954. Could you take a look at those? @apisandipas did most of that thinking, so we're happy to review this PR, just want to make sure we are all on the same page about expectations / ACs.

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+    it('navigates to PACT Act page - C32630', () => {
+      cy.get('.homepage-hero')
+        .scrollIntoView()
+        .within(() => {
+          cy.contains('Learn what the PACT Act means for you', {
+            selector: 'a',
+          }).click();
+          cy.location('pathname').should(
+            'eq',
+            '/resources/the-pact-act-and-your-va-benefits/',
```
Comment (apisandipas) [inline src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:89]: Dynamic content assertion

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+        checkLinkNavigation(
+          linksListSelector,
+          'Get reimbursed for travel pay',
+          '/health-care/get-reimbursed-for-travel-pay/',
+        );
+        checkLinkNavigation(
+          linksListSelector,
+          'Get your VA medical records',
+          '/health-care/get-medical-records/',
+        );
```
Comment (apisandipas) [inline src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:294]: Dynamic content assertion

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+        cy.get('.homepage-blog')
+          .should('exist')
+          .scrollIntoView()
+          .within(() => {
+            cy.get('h3#pathfinder-the-front-door-for-')
+              .should('exist')
+              .next('p')
+              .should('exist')
+              .and('not.be.empty');
+          });
```
Comment (apisandipas) [inline src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:313]: Dynamic content assertion

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+        checkLinkNavigation(
+          '.homepage-blog',
+          'Pathfinder: The front door for engaging with VA',
+          'https://pathfinder.va.gov/',
+        );
+        checkLinkNavigation(
+          '.homepage-blog',
+          'Read the full article',
+          'https://pathfinder.va.gov/',
+        );
```
Comment (apisandipas) [inline src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:328]: Dynamic content assertion

File: src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+      checkHubContents('family-member-benefits');
+      checkHubContents('burials-and-memorials');
+      checkHubContents('careers-and-employment');
+      checkHubContents('housing-assistance');
+      checkHubContents('pension');
+      checkHubContents('life-insurance');
+      checkHubContents('education-and-training');
+      checkHubContents('records');
+      checkHubContents('health-care');
+      checkHubContents('disability');
```
Comment (apisandipas) [inline src/applications/static-pages/homepage/tests/new-homepage.cypress.spec.js:361]: Dynamic content assertion. These may change, but typically far less often. There is also another one to be added in the near future called VA Department Information.

**Review by apisandipas — DISMISSED** [review]

**Comment by tlei123** [comment]
Thx @jilladams @apisandipas for checking this!  FYI, I'm gonna just lay off this a day or two...

This spec may be handed off to your Public Websites team, depending on contract-decisions affecting my PTEMS Shared Support team.  I'll ping you here once I find out if I'm still on-contract for this, hopefully by end of tmrw (Tue 20th).

**Comment by tlei123** [comment]
@jilladams @apisandipas , sorry!  I JUST found out our entire PTEMS Shared Support team's supposed to just take PTO tmrw (Tue 20th) through EoY &mdash; can't do billable work.  :(

@apisandipas , IF you'd still like me to hammer at this today, just ping me right away on Slack [**Tze Lei** (Ad Hoc workspace) or **Tze-chiu Lei** (DSVA workspace)].  Otherwise, just take over my branch here and change as desired/needed.  I'll circle back next year if I get on a permissible contract for this.

cc @mmiddaugh

**Comment by jilladams** [comment]
Thanks for the heads up. Noted, and we'll go forward from here. Thank you for your contrib!!!

**Review by it-harrison — APPROVED** [review]

**Pull request merged**